### ISSUE TITLE
FileNotFoundError when building on Windows

### ISSUE THREAD

**Issue by Yao1993** [dave-msk/merak#4]
When building the exmaple on Windows,  FileNotFoundError occured.

```
(pybinary) ➜  merak git:(master) merak cythonize examples/foo foo-build
Traceback (most recent call last):
  File "c:\anaconda3\envs\pybinary\lib\shutil.py", line 791, in move
    os.rename(src, real_dst)
FileNotFoundError: [WinError 3] 系统找不到指定的路径。: 'C:\\Users\\tm42h7\\AppData\\Local\\Temp\\tmprmcrvgn4\\foo\\bar\\baz.py' -> 'C:\\Users\\tm42h7\\AppData\\Local\\Temp\\tmprmcrvgn4\\foo\\bar\\___bar\\baz.py'

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "c:\anaconda3\envs\pybinary\lib\runpy.py", line 194, in _run_module_as_main
    return _run_code(code, main_globals, None,
  File "c:\anaconda3\envs\pybinary\lib\runpy.py", line 87, in _run_code
    exec(code, run_globals)
  File "C:\Anaconda3\envs\pybinary\Scripts\merak.exe\__main__.py", line 7, in <module>
  File "c:\anaconda3\envs\pybinary\lib\site-packages\merak\main.py", line 34, in main
    app.run()
  File "c:\anaconda3\envs\pybinary\lib\site-packages\cement\core\foundation.py", line 916, in run
    return_val = self.controller._dispatch()
  File "c:\anaconda3\envs\pybinary\lib\site-packages\cement\ext\ext_argparse.py", line 808, in _dispatch
    return func()
  File "c:\anaconda3\envs\pybinary\lib\site-packages\merak\commands\cythonize.py", line 57, in __call__
    success = cybuild.build_package_cython_extension(
  File "c:\anaconda3\envs\pybinary\lib\site-packages\merak\core\cybuild.py", line 59, in build_package_cython_extension
    mods, sub_pkgs = _restructure_package(tmp_proot, sep=sep)
  File "c:\anaconda3\envs\pybinary\lib\site-packages\merak\core\cybuild.py", line 179, in _restructure_package
    project.do(rename_cs)
  File "c:\anaconda3\envs\pybinary\lib\site-packages\rope\base\project.py", line 113, in do
    self.history.do(changes, task_handle=task_handle)
  File "c:\anaconda3\envs\pybinary\lib\site-packages\rope\base\history.py", line 36, in do
    changes.do(change.create_job_set(task_handle, changes))
  File "c:\anaconda3\envs\pybinary\lib\site-packages\rope\base\change.py", line 65, in do
    change.do(job_set)
  File "c:\anaconda3\envs\pybinary\lib\site-packages\rope\base\change.py", line 126, in call
    function(self)
  File "c:\anaconda3\envs\pybinary\lib\site-packages\rope\base\change.py", line 201, in do
    self._operations.move(self.resource, self.new_resource)
  File "c:\anaconda3\envs\pybinary\lib\site-packages\rope\base\change.py", line 339, in move
    fscommands.move(resource.real_path, new_resource.real_path)
  File "c:\anaconda3\envs\pybinary\lib\site-packages\rope\base\fscommands.py", line 45, in move
    shutil.move(path, new_location)
  File "c:\anaconda3\envs\pybinary\lib\shutil.py", line 811, in move
    copy_function(src, real_dst)
  File "c:\anaconda3\envs\pybinary\lib\shutil.py", line 435, in copy2
    copyfile(src, dst, follow_symlinks=follow_symlinks)
  File "c:\anaconda3\envs\pybinary\lib\shutil.py", line 264, in copyfile
    with open(src, 'rb') as fsrc, open(dst, 'wb') as fdst:
FileNotFoundError: [Errno 2] No such file or directory: 'C:\\Users\\tm42h7\\AppData\\Local\\Temp\\tmprmcrvgn4\\foo\\bar\\___bar\\baz.py'
```

My  environment is:
```
(pybinary) ➜  merak git:(master) conda list
# packages in environment at C:\Anaconda3\envs\pybinary:
#
# Name                    Version                   Build  Channel
ca-certificates           2021.7.5             haa95532_1
cement                    3.0.4                    pypi_0    pypi
certifi                   2021.5.30        py38haa95532_0
colorama                  0.4.4                    pypi_0    pypi
colorlog                  4.8.0                    pypi_0    pypi
cython                    0.29.24                  pypi_0    pypi
merak                     0.2.3                    pypi_0    pypi
openssl                   1.1.1k               h2bbff1b_0
pip                       21.1.3           py38haa95532_0
python                    3.8.10               hdbf39b2_7
rope                      0.19.0                   pypi_0    pypi
setuptools                51.3.3                   pypi_0    pypi
sqlite                    3.36.0               h2bbff1b_0
vc                        14.2                 h21ff451_1
vs2015_runtime            14.27.29016          h5e58377_2
wheel                     0.36.2             pyhd3eb1b0_0
wincertstore              0.2                      py38_0
```

**Comment by dave-msk** [comment]
I haven't tried that in Windows yet. I'll get back to you soon

**Comment by dave-msk** [comment]
The target path seems to have some problem in Windows. I'll take a look at this weekend

**Comment by dave-msk** [comment]
Turns out `rope` uses `/` as path separator instead of the environment-defined one. I've corrected the file restructuring behavior. Please update to v0.2.4 and try again.

**Comment by Yao1993** [comment]
After updating merak to v0.2.4, there was a link error.
```
(pybinary) ➜  merak git:(master) ✗ merak cythonize examples/foo foo-build -v
2021-07-29 00:21:37,245 (INFO) merak.commands.cythonize : Building binary package ...
2021-07-29 00:21:37,247 (INFO) merak.core.cybuild : 1. Copying package to temporary directory ...
2021-07-29 00:21:37,251 (INFO) merak.core.cybuild : 2. Restructuring package ...
2021-07-29 00:21:37,251 (INFO) merak.core.cybuild : 2.1. Restructuring Python modules ...
2021-07-29 00:21:37,295 (INFO) merak.core.cybuild : 2.1. Done!
2021-07-29 00:21:37,295 (INFO) merak.core.cybuild : 2.2. Injecting finder to main `__init__` ...
2021-07-29 00:21:37,296 (INFO) merak.core.cybuild : 2.2. Done!
2021-07-29 00:21:37,296 (INFO) merak.core.cybuild : 3. Adding Cython setup file ...
2021-07-29 00:21:37,300 (INFO) merak.core.cybuild : 3. Done!
2021-07-29 00:21:37,300 (INFO) merak.core.cybuild : 4. Compiling package binary ...
2021-07-29 00:21:40,061 (ERROR) merak.utils.subproc : error: command 'C:\\Program Files (x86)\\Microsoft Visual Studio\\2019\\Community\\VC\\Tools\\MSVC\\14.29.30037\\bin\\HostX86\\x64\\link.exe' failed with exit status 1104
2021-07-29 00:21:40,097 (ERROR) merak.core.cybuild : Failed to compile package!
```

I found a solution from [how-to-compile-init-py-file-using-cython-on-windows](https://stackoverflow.com/questions/58797673/how-to-compile-init-py-file-using-cython-on-windows.).
I opened a pull request #5.

**Comment by Yao1993** [comment]
Ops! ImportError occured when trying to import foo.

```python 
>>> import foo
<stdin>:1: DeprecationWarning: Deprecated since Python 3.4. Use importlib.util.find_spec() instead.
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "foo\__init__.py", line 88, in init foo.__init__
  File "foo\___bar.py", line 5, in init foo.___bar
ImportError: cannot import name ___bar_baz
```
I checked the pyd file and found there was no `PyInit___bar_baz`.
```
(pybinary) C:\Users\tm42h7\Desktop\debug\merak\foo-build\foo>dumpbin /EXPORTS __init__.cp38-win_amd64.pyd
Microsoft (R) COFF/PE Dumper Version 14.16.27045.0
Copyright (C) Microsoft Corporation.  All rights reserved.


Dump of file __init__.cp38-win_amd64.pyd

File Type: DLL

  Section contains the following exports for __init__.cp38-win_amd64.pyd

    00000000 characteristics
    FFFFFFFF time date stamp
        0.00 version
           1 ordinal base
           2 number of functions
           2 number of names

    ordinal hint RVA      name

          1    0 0000DF80 PyInit____bar
          2    1 00005E60 PyInit_foo

  Summary

        4000 .data
        1000 .pdata
        3000 .rdata
        1000 .reloc
        1000 .rsrc
        E000 .text
```

**Comment by dave-msk** [comment]
Merged #5 . Please update to `v0.2.5` for the changes.

**Comment by Yao1993** [comment]
Ok, it works fine.
